src/assets/scripts/client/aircraft/AircraftController.js

```diff
@@ -568,6 +641,80 @@ export default class AircraftController {
     }
 
     /**
+     * Generate a unique `transponderCode`
+     *
+     * This method should only be run while building props for a
+     * soon-to-be-instantiated `AircraftModel`
+     *
+     * @for AircraftController
+     * @method _generateUniqueTransponderCode
+     * @return {number}
+     * @private
+     */
+    _generateUniqueTransponderCode() {
+        const minCodeValue = 1203;
+        const maxCodeValue = 7499;
+        const transponderCode = _random(minCodeValue, maxCodeValue);
+
+        if (!this._isValidTransponderCode(transponderCode)) {
+            // the value generated is already in use, recurse back through this method and try again
+            this._generateUniqueTransponderCode();
+        }
+
+        this._addTransponderCodeToInUse(transponderCode);
+
+        return transponderCode;
+    }
+
+    /**
+     * Add a given `transponderCode` to the `#_transponderCodesInUse` list
+     *
+     * @for AircraftController
+     * @method _addTransponderCodeToInUse
+     * @param transponderCode {number}
+     */
+    _addTransponderCodeToInUse(transponderCode) {
+        this._transponderCodesInUse.push(transponderCode);
+    }
+
+    /**
+     * Remove the `transponderCode` from the list of `#_transponderCodesInUse`
+     *
+     * @for AircraftController
+     * @method _removeTransponderCodeFromUse
+     * @param transponderCode {number}
+     */
+    _removeTransponderCodeFromUse({ transponderCode }) {
+        this._transponderCodesInUse = _without(this._transponderCodesInUse, transponderCode);
+    }
+
+    /**
+     * Boolean helper used to determine if a given `transponderCode` is already
+     * present within the `#_transponderCodesInUse` list.
+     *
+     * @for AircraftController
+     * @method _isTransponderCodeInUse
+     * @param transponderCode {number}
+     * @return {booelean}
+     */
+    _isTransponderCodeInUse(transponderCode) {
+        return this._transponderCodesInUse.indexOf(transponderCode) !== -1;
+    }
+
+    /**
+     * Boolean helper used to determine if a given `transponderCode` is both
+     * not in use and not an invalid transponderCode.
+     *
+     * @for AircraftController
+     * @method _isValidTransponderCode
+     * @param transponderCode {number}
+     * @return {boolean}
+     */
+    _isValidTransponderCode(transponderCode) {
+        return !this._isTransponderCodeInUse(transponderCode) && INVALID_SQUAWK_CODES.indexOf(transponderCode) === -1;
```

Comment:
Correct. Valid transponder codes may only use digits 0 through 7. Though I would say the `INVALID_SQUAWK_CODES` should be renamed to `RESERVED_SQUAWK_CODES` for clarity, because they are not inherently "invalid", they just aren't what we can assign aircraft for purposes of tracking them: "discrete codes" (aka unique, non-reserved codes).